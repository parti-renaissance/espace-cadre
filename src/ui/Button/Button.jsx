import { Button as MuiButton } from '@mui/material'
import PropTypes from 'prop-types'
import { styled } from '@mui/system'
import { shouldForwardProps } from 'components/shared/shouldForwardProps'
import Loader from 'ui/Loader'

const MainButton = styled(
  MuiButton,
  shouldForwardProps
)(
  ({ theme, isMainButton }) => `
  color: ${theme.palette.button.color.main};
  background: ${isMainButton ? 'rgba(55, 67, 200, 0.08)' : theme.palette.button.background.main};
  &:hover {
    background: ${isMainButton ? 'rgba(0, 0, 0, 0.08)' : theme.palette.button.background.main};
  }
  &.Mui-disabled {
    color: ${theme.palette.button.color.disabled};
    background-color: ${theme.palette.button.background.disabled};
  }
  height: 35px;
  border-radius: 8px;
  border: none;
  padding: ${theme.spacing(0.75, 2)};
`
)

const Button = ({ children, onClick, rootProps, disabled = false, isMainButton = false }) => (
  <MainButton variant="contained" onClick={onClick} {...rootProps} disabled={disabled} isMainButton={isMainButton}>
    {children}
  </MainButton>
)

export const ActionButton = ({ children, handleSubmit, isLoading = false }) => (
  <Button disabled={isLoading} onClick={handleSubmit} rootProps={{ sx: { color: 'whiteCorner', mr: 2 } }}>
    {isLoading && <Loader />}&nbsp;
    {children}
  </Button>
)

export default Button

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
  rootProps: PropTypes.object,
  disabled: PropTypes.bool,
  isMainButton: PropTypes.bool,
}

ActionButton.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  handleSubmit: PropTypes.func,
  rootProps: PropTypes.object,
}
