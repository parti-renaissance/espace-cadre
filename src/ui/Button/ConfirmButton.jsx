import { IconButton } from '@mui/material'
import { useState } from 'react'
import PropTypes from 'prop-types'
import ConfirmationModal from 'ui/Confirmation/ConfrmationModal'

const ConfirmButton = ({ title, description, children, onClick }) => {
  const [open, setOpen] = useState(false)

  return (
    <>
      <IconButton edge="start" color="inherit" onClick={() => setOpen(true)} aria-label="delete" sx={{ ml: 0.5 }}>
        {children}
      </IconButton>
      {open && (
        <ConfirmationModal
          onCancel={() => setOpen(false)}
          onConfirm={() => {
            onClick()
            setOpen(false)
          }}
          title={title}
          description={description}
        />
      )}
    </>
  )
}

export default ConfirmButton

ConfirmButton.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
}
