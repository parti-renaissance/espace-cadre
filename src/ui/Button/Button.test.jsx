import { fireEvent, render, screen } from '@testing-library/react'
import Button from './Button'

vi.mock('@mui/system', () => ({
  styled: c => () => c,
}))

vi.mock('@mui/material', () => ({
  Button: ({ children, isMainbutton, ...props }) => (
    <div data-testid="mui-button-mock" {...props}>
      {children}
    </div>
  ),
}))

describe('Button', () => {
  it('displays button', () => {
    const mock = vi.fn()
    const { container } = render(
      <Button onClick={mock} rootProps={{ sx: 'foo' }}>
        Foo
      </Button>
    )

    expect(container).toMatchSnapshot()
  })
  it('calls handleclick on click', () => {
    const mockOnClick = vi.fn()
    render(<Button onClick={mockOnClick}>Foo</Button>)

    const button = screen.getByTestId('mui-button-mock')
    fireEvent.click(button)
    expect(mockOnClick).toHaveBeenCalled()
  })
})
