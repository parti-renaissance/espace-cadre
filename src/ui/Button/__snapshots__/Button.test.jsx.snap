// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > displays button 1`] = `
<div>
  <div
    data-testid="mui-button-mock"
    sx="foo"
    variant="contained"
  >
    Foo
  </div>
</div>
`;
